htlcswitch/hop/payload_test.go

```diff
@@ -348,6 +348,81 @@ var decodePayloadTests = []decodePayloadTest{
 		},
 		shouldHaveTotalAmt: true,
 	},
+	{
+		name:       "final blinded hop with total amount",
+		isFinalHop: true,
+		payload: []byte{
+			// amount
+			0x02, 0x00,
+			// cltv
+			0x04, 0x00,
+			// encrypted data
+			0x0a, 0x03, 0x03, 0x02, 0x01,
+		},
+		shouldHaveEncData: true,
+	},
+	{
```

Comment:
Nit: While checking the code coverage, I found that the new validation where an intermediate hop has a zero chanID is not part of this test, but maybe it's tested in the routing pkg. 

<img width="841" alt="Screenshot 2023-12-16 at 11 05 42" src="https://github.com/lightningnetwork/lnd/assets/90319354/9605403a-e5d6-44e1-942d-cc039b32ad5b">